Portal/src/Datahub.Portal/Middleware/WorkspaceAppMiddleware.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Security.Claims;
+
+namespace Datahub.Portal.Middleware;
+
+public class WorkspaceAppMiddleware
+{
+    private readonly RequestDelegate _next;
+    private readonly bool _enabled;
+
+    public WorkspaceAppMiddleware(RequestDelegate next, IConfiguration config)
+    {
+        _next = next;
+        _enabled = config.GetValue<bool>("ReverseProxy:Enabled");
+    }
+
+    public async Task InvokeAsync(HttpContext context)
+    {
+        if (_enabled)
+        {
+            var urlPath = context.Request.Path;
+            if (urlPath.StartsWithSegments("/wsapp", StringComparison.OrdinalIgnoreCase))
```

Comment:
Can we get this path into a constant?

Reply:
Make sense, I'll read it from the config.